src/shady/parser/token.c

```diff
@@ -132,6 +134,13 @@ Token next_token(Tokenizer* tokenizer) {
         while (in_bounds(tokenizer, token_size) && is_digit(slice[token_size])) {
             token_size++;
         }
+        if (slice[token_size] == 'e')
+            token_size++;
+        if (slice[token_size] == '-' || slice[token_size] == '+')
```

Comment:
this exponent sign and value can only come after 'e' right ? if so it should be predicated by that

Reply:
you are right. this is addressed in 0f19bdc